api/types.proto

```diff
@@ -849,6 +849,37 @@ message SecretReference {
 	}
 }
 
+// ConfigReference is the linkage between a service and a config that it uses.
+message ConfigReference {
+	// ConfigID represents the ID of the specific Config that we're
+	// referencing.
+	string config_id = 1;
+
+	// ConfigName is the name of the config that this references, but this is just provided for
+	// lookup/display purposes. The config in the reference will be identified by its ID.
+	string config_name = 2;
+
+	// FileTarget represents a specific target that is backed by a file
+	message FileTarget {
```

Comment:
> Can we make `FileTarget` a higher-level object and reuse it between `SecretReference` and `ConfigReference`?

I'll do this if they're still the same when we're ready to merge this PR, but I want to leave open the option of changing the config-specific `FileTarget` to suit configs. For example, we might want split out the directory into a different field from the file basename.